### ISSUE TITLE
Error in $: Failed reading: not a valid json value

### ISSUE THREAD

**Issue by Tharre** [LaurentRDC/pandoc-pyplot#2]
Hey there,

I've been trying to run the example from usage, but it's throwing an error:

```
$ pandoc --filter pandoc-pyplot input.md --output output.pdf
Error running filter pandoc-pyplot:
Error in $: Failed reading: not a valid json value
```

Any idea what might cause this?

My system is x86_64 Arch Linux with pandoc version 2.7.2. I've installed pandoc-pyplot via stack.

Thanks in advance.

**Comment by LaurentRDC** [comment]
Hi Tharre,

The way pandoc-pyplot and pandoc communicate is via JSON. As you can see, somewhere this communication is breaking down. I can see two potential problems:

1. I have not tried pandoc-pyplot and pandoc 2.7 together. This interaction might be different than with pandoc 2.5 (which I use for testing), although I can't see why that would be the case.
2. There is a character in your input.md that trips pandoc-pyplot for some reason. Are you using the README example?

```markdown
    ```{.pyplot}
    import matplotlib.pyplot as plt

    plt.figure()
    plt.plot([0,1,2,3,4], [1,2,3,4,5])
    plt.title('This is an example figure')
    ```
```

I will try the above example with pandoc 2.7.2

**Comment by LaurentRDC** [comment]
It is due to pandoc-pyplot's input arguments. Pandoc-pyplot was getting confused between inputs from pandoc and invalid flags.

This behavior is not present in pandoc-pyplot v2.1.0.1. You should be able to use it without problems. I will release an update to fix it this morning (v2.1.1.1).

**Edit**: a new version with the appropriate fix has been [released](http://hackage.haskell.org/package/pandoc-pyplot-2.1.1.1). Give it a try and let me know if this is fixed.

Forgive me for the inconvenience!

**Comment by Tharre** [comment]
Can confirm that it works.

Thanks!
